hail/src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -1042,7 +1043,8 @@ class IRSuite extends SparkSuite {
       MatrixMultiWrite(Array(mt, mt), MatrixNativeMultiWriter(tmpDir.createLocalTempFile())),
       MatrixAggregate(mt, MakeStruct(Seq("foo" -> count))),
       BlockMatrixWrite(blockMatrix, tmpDir.createLocalTempFile(), false, false, false),
-      CollectDistributedArray(ArrayRange(0, 3, 1), 1, "x", "y", Ref("x", TInt32()))
+      CollectDistributedArray(ArrayRange(0, 3, 1), 1, "x", "y", Ref("x", TInt32())),
+      ReadPartition(Str("foo"), CodecSpec.default, TStruct("foo"->TInt32(), "bar" -> TString()), TStruct("foo"->TInt32()))
```

Comment:
What other examples are there for `path` where it's not a `Str`? Why not have a String?

Reply:
I want to be able to treat a TableRead as essentially an array of `ReadPartition`s---This means I want to do something that effectively looks like:
```
CollectDistributedArray(
  MakeArray(FastSeq(Str("p1"), ....), TArray(TString())), \\ the contexts are just the paths
  MakeStruct(FastSeq()), \\ no globals
  "context", "globals",
  partition_f(ReadPartition(Ref("context", TString()), ....))
)
```
where I compile one function to handle all the partitions which takes the path (from each partition) as input.

Reply:
I see, sounds good. Thanks!